blog/2020-05-08-AIVersusCovid-19.adoc

```diff
@@ -0,0 +1,135 @@
+= AI versus Covid-19: How Java helps nurses and doctors in this fight
+:page-interpolate: true
+:awestruct-author: Christopher-Chianelli
+:awestruct-layout: blogPostBase
+:awestruct-tags: [use case]
+
+Scheduling a hospital is tough. To maintain employees' happiness and health,
+you need to take into account sick days, days off requests, and assure
+the employees have ample time between shifts. This is doubly so in a pandemic,
+where you need to minimize exposure to the new disease and have nurses with
+different risks. Let's take a look at how we can quickly and efficiently
+schedule a hospital roster that minimizes cross-contamination between wards
+and maximizes employees' happiness using OptaWeb Employee Rostering.
+
+== The domain
+
+First, let's set out the domain of our model:
+
+* There are _wards_, which are sections of the hospital, and are separated into
+Covid-19 wards and non-Covid-19 wards
+
+* There are _employees_, who are doctors and nurses who can be assigned to a ward,
+each having different susceptibilities to Covid-19 (for instance,
+some nurses are older and thus are more vulnerable)
+
+* There are _skills_, which describes what skill set an employee must have in
+order to work in a particular ward
+
+* There are _shifts_, which are positions that need to be filled at a given ward
+at a particular time, and may require additional skills in addition to those of
+its ward (for example, we need a repository specialist nurse in a Covid-19
+ward, as well as a nurse and a doctor)
+
+* Finally, there are _availabilities_, which describes when an employee is unable to
+work, desires to work, or is requesting time off
+
+The following class diagram shows the Java classes that represent this domain:
+
+image::CovidEmployeeRosteringClassDiagram.png[Class Diagram]
+
+== The constraints
+
+OptaWeb Employee Rostering comes with a set of builtin
+constraints, which you can add to, modify or remove according to your needs.
+A few examples:
+
+* An employee cannot work two shifts that overlap; after all, how do you expect an
+employee to be in two places at the same time? As such, this is a hard constraint,
+meaning a schedule is infeasible if broken.
+
+* Whenever possible, an employee should not be working on a day they request off.
+Since the schedule remains feasible an employee is working on a day they want off,
+this is a soft constraint.
+
+* The higher an employee's Covid-19 susceptibility is, the less they should
+work in a Covid-19 ward. This is another soft constraint.
+
+These, and the rest of the constraints, can be found in the
+https://github.com/kiegroup/optaweb-employee-rostering/blob/covid-19/optaweb-employee-rostering-backend/src/main/resources/org/optaweb/employeerostering/service/solver/employeeRosteringScoreRules.drl[Drools constraints
+file]. They can also be implemented in Java using constraint streams.
```

Comment:
Very soft suggestion: _... in Java using OptaPlanner Constraint Stream API_ or something like that. Why? To avoid Java non-experts thinking: Wow, I didn't know Java had constraint streams! What's that?